gw_eccentricity/eccDefinition.py

```diff
@@ -251,16 +296,25 @@ def __init__(self, dataDict, num_orbits_to_exclude_before_merger=2,
         self.t_for_omega22_average = None
         self.omega22_average = None
 
-        if "hlm_zeroecc" in self.dataDict:
-            self.compute_res_amp_and_omega22()
+        # compute residual data
+        if any(["hlm_zeroecc" in self.dataDict,
+                "amplm_zeroecc" in self.dataDict and "omegalm_zeroecc" in self.dataDict,
+                "amplm_zeroecc" in self.dataDict and "phaselm_zeroecc" in self.dataDict]):
```

Comment:
Just check for `hlm_zeroecc` and `amplm_zeroecc`? Why do you care here whether phase or omega is given?

Reply:
Because we can compute residual omega22 if omega22_zeroecc is provided? Note: We are computing both omega22 residual and amp22 residual in the following function.